databricks/koalas/generic.py

```diff
@@ -1632,6 +1634,136 @@ def first_valid_index(self):
+        rabbit    4    5      6
+        """
+
+        if sum(x is not None for x in (items, like, regex)) > 1:
+            raise TypeError(
+                "Keyword arguments `items`, `like`, or `regex` " "are mutually exclusive"
+            )
+
+        is_series = isinstance(self, ks.Series)
+
+        kdf = self
+        if is_series:
+            kdf = self.to_frame()
+            axis = validate_axis(axis, none_axis=0)
+            if isinstance(kdf, ks.Series) and axis == 1:
+                raise ValueError("Series does not support columns axis.")
+        else:
+            axis = validate_axis(axis, none_axis=1)
+        assert isinstance(kdf, ks.DataFrame)
+
+        index_scols = kdf._internal.index_spark_columns
+
+        if items is not None:
+            if is_list_like(items):
+                items = list(items)
+            else:
+                raise ValueError("items should be a list-like object.")
+            if axis == 0:
+                # TODO: support multi-index here
+                if len(index_scols) != 1:
+                    raise ValueError("Single index must be specified.")
+                col = None
+                for item in items:
+                    if col is None:
+                        col = index_scols[0] == F.lit(item)
+                    else:
+                        col = col | (index_scols[0] == F.lit(item))
+                filtered_df = ks.DataFrame(kdf._internal.with_filter(col))
+            elif axis == 1:
+                filtered_df = kdf[items]
+        elif like is not None:
+            if axis == 0:
+                # TODO: support multi-index here
+                if len(index_scols) != 1:
+                    raise ValueError("Single index must be specified.")
+                filtered_df = ks.DataFrame(kdf._internal.with_filter(index_scols[0].contains(like)))
+            elif axis == 1:
+                column_labels = kdf._internal.column_labels
+                output_labels = [label for label in column_labels if any(like in i for i in label)]
+                filtered_df = kdf[output_labels]
+        elif regex is not None:
+            if axis == 0:
+                # TODO: support multi-index here
+                if len(index_scols) != 1:
+                    raise ValueError("Single index must be specified.")
+                filtered_df = ks.DataFrame(kdf._internal.with_filter(index_scols[0].rlike(regex)))
+            elif axis == 1:
+                column_labels = kdf._internal.column_labels
+                matcher = re.compile(regex)
+                output_labels = [
+                    label
+                    for label in column_labels
+                    if any(matcher.search(i) is not None for i in label)
+                ]
+                filtered_df = kdf[output_labels]
+        else:
+            raise TypeError("Must pass either `items`, `like`, or `regex`")
+
+        if is_series:
+            return filtered_df.iloc[:, 0]
```

Comment:
If you wanna get a first `Series` from the `DataFrame`, let's use `first_series`.

you can

```python
from databricks.koalas.series import first_series
first_series(filtered_df)
```